command/windows_service_install_test.go

```diff
@@ -0,0 +1,361 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package command
+
+import (
+	"bytes"
+	"errors"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+	"text/template"
+
+	"github.com/hashicorp/cli"
+	"github.com/hashicorp/nomad/helper/winsvc"
+	"github.com/shoenig/test/must"
+)
+
+func TestWindowsServiceInstallCommand_Run(t *testing.T) {
+	t.Parallel()
+
+	freshInstallFn := func(m *winsvc.MockWindowsServiceManager) {
+		srv := winsvc.NewMockWindowsService(t)
+		m.ExpectIsServiceRegistered(winsvc.WINDOWS_SERVICE_NAME, false, nil)
+		m.ExpectIsServiceRegistered(winsvc.WINDOWS_SERVICE_NAME, false, nil)
+		m.ExpectCreateService(winsvc.WINDOWS_SERVICE_NAME, "nomad.exe",
+			winsvc.WindowsServiceConfiguration{}, srv, nil)
+		srv.ExpectEnableEventlog(nil)
+		srv.ExpectStop(nil)
+		srv.ExpectStart(nil)
+	}
+	upgradeInstallFn := func(m *winsvc.MockWindowsServiceManager) {
+		srv := winsvc.NewMockWindowsService(t)
+		m.ExpectIsServiceRegistered(winsvc.WINDOWS_SERVICE_NAME, true, nil)
+		m.ExpectIsServiceRegistered(winsvc.WINDOWS_SERVICE_NAME, true, nil)
+		m.ExpectGetService(winsvc.WINDOWS_SERVICE_NAME, srv, nil)
+		m.ExpectGetService(winsvc.WINDOWS_SERVICE_NAME, srv, nil)
+		srv.ExpectConfigure(winsvc.WindowsServiceConfiguration{}, nil)
+		srv.ExpectEnableEventlog(nil)
+		srv.ExpectStop(nil)
+		srv.ExpectStop(nil)
+		srv.ExpectStart(nil)
+	}
+
+	testCases := []struct {
+		desc        string
+		args        []string
+		privilegeFn func() bool
+		setup       func(string, *winsvc.MockWindowsServiceManager)
+		after       func(string)
+		output      string
+		errOutput   string
+		status      int
+	}{
+		{
+			desc: "fresh install success",
+			setup: func(_ string, m *winsvc.MockWindowsServiceManager) {
+				freshInstallFn(m)
+			},
+			output: "Success",
+			status: 0,
+		},
+		{
+			desc: "fresh install writes config",
+			setup: func(_ string, m *winsvc.MockWindowsServiceManager) {
+				freshInstallFn(m)
+			},
+			after: func(dir string) {
+				must.FileExists(t, filepath.Join(dir, "programdata/HashiCorp/nomad/config/config.hcl"))
```

Comment:
similar deal here and in some of these `setup:`s -- if this fails, where does the error output point to for debugging?  I think these ones are ok, just checkin.

Reply:
It will fail with that specific line, and it will also show the test name as failing.

```
...
    windows_service_install_test.go:70:
         windows_service_install_test.go:70: expected file to exist
         ↪  name: /tmp/nomad-test1896632722/programdata/HashiCorp/nomad/config/config.hcl.txt
         ↪ error: stat /tmp/nomad-test1896632722/programdata/HashiCorp/nomad/config/config.hcl.txt: no such file or directory
...
     --- FAIL: TestWindowsServiceInstallCommand_Run/fresh_install_writes_config (0.06s)